isa/rv64mi/breakpoint.S

```diff
@@ -97,6 +97,51 @@ RVTEST_CODE_BEGIN
   lw a2, (a2)
   bnez a2, fail
 
+2:
+  # Test that address breakpoints have higher priority than unaligned loads.
+  li TESTNUM, 12
+  li a0, (2 << (__riscv_xlen - 4)) | MCONTROL_M | MCONTROL_LOAD | MCONTROL_STORE
+  csrw tdata1, a0
+  # Skip if breakpoint type is unsupported.
+  csrr a1, tdata1
+  bne a0, a1, 2f
+  la a2, data1 + 1
+  la a3, data3
+  csrw tdata2, a2
+
+  # Slow path because CSR was *just* written. (This is spike-specific.)
+  lw a2, (a2)
+  beqz a2, fail
+
+  # Normal aligned load, so that the next time we'll hit the fast path.
+  li TESTNUM, 13
+  lw a0, (a3)
+
+  # Fast path
+  li TESTNUM, 14
+  lw a2, (a2)
+  beqz a2, fail
+
+  # Test that address breakpoints have higher priority than unaligned stores.
+  li TESTNUM, 16
+
+  la a2, data1 + 1
+  csrw tdata2, a2
+
+  # Slow path because CSR was *just* written. (This is spike-specific.)
+  sw x0, (a2)
+  beqz a2, fail
```

Comment:
Presumably this branch should be removed since the store can't possibly write `x2`?

Reply:
Good point. Definitely a cut and paste issue. I just removed the checks, since if the trigger didn't match then you end up with an unaligned exception which will cause the test to fail.